xwakes/wit/component.py

```diff
@@ -385,3 +510,895 @@ def get_shorthand_type(self) -> str:
+        :param exponents: four integers corresponding to (source_x, source_y,
+        test_x, test_y) aka (a, b, c, d)
+        :param layer: the chamber material, as a wit Layer object
+        :param radius: the chamber radius in m
+        :return: A component object
+        """
+
+        if (layer is None and resistivity is None):
+            raise ValueError(
+                "Either a layer object or a resistivity must be provided")
+        if layer is not None and resistivity is not None:
+            raise ValueError(
+                "Both a layer object and a resistivity cannot be provided")
+
+        if radius is None:
+            raise ValueError("Radius must be provided")
+
+        self.layer = layer
+        self.radius = radius
+        self.plane = plane
+        self.factor = factor
+        self.length = length
+        self.resistivity = resistivity
+        self.zero_rel_tol = zero_rel_tol
+
+        source_exponents, test_exponents, plane = _handle_plane_and_exponents_input(
+                                    kind=kind, exponents=exponents,
+                                    source_exponents=source_exponents,
+                                    test_exponents=test_exponents,
+                                    plane=plane)
+
+        # we set impedance and wake to a dummy callable because they will be
+        # overridden by methods
+        super().__init__(impedance=lambda x: 0, wake=lambda x: 0, plane=plane,
+                         source_exponents=source_exponents,
+                         test_exponents=test_exponents,
+                         name="Classic Thick Wall")
+
+    def impedance(self, f):
+        layer = self.layer
+        material_magnetic_permeability = mu0
+        if layer is not None:
+            self._check_layer(layer) # Checks that there are no unsupported layer properties
+                                # (e.g. permeability different from vacuum)
+            material_resistivity = layer.dc_resistivity
+        else:
+            material_resistivity = self.resistivity
+
+        radius = self.radius
+        plane = self.plane
+        factor = self.factor
+        length = self.length
+        exponents = tuple(self.source_exponents + self.test_exponents)
+
+        if plane == 'z' and exponents == (0, 0, 0, 0):
+            out = factor * length * ((1 + np.sign(f)*1j) * material_resistivity
+                            / (2* np.pi * radius)
+                            / self.delta_skin(f, material_resistivity,
+                                              material_magnetic_permeability))
+        # Transverse dipolar impedance
+        elif ((plane == 'x' and exponents == (1, 0, 0, 0)) or
+                (plane == 'y' and exponents == (0, 1, 0, 0))):
+            out = factor * length * ((1 + np.sign(f)*1j) * material_resistivity
+                            / (np.pi * radius**3 * (2 * np.pi * f * np.sqrt(eps0 * mu0)))
```

Comment:
c = 1/sqrt(eps0*mu0), hence I would add the factor `c_light` in front, and remove `np.sqrt(eps0 * mu0)` (as it was... but maybe there was a reason to do it this way?)

Reply:
No good reason to do that I think. I will change it